### PR TITLE
Integrate geographies filter 

### PR DESCRIPTION
**Pull request by THOR300**
# Description

This pull request updates the `Filters` object and relating `YQL` and `query` code to enable filtering for `family_geographies`. 

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Skip auto-tagging
- [ ] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail (integrated soon)
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [x] If this PR fixes a bug, I've added a test that will fail without my fix.
- [x] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

### CODE REVIEW

File: src/cpr_sdk/yql_builder.py
Comment (olaughter) [inline src/cpr_sdk/yql_builder.py]: `contains` is a string operator in vespa. Won't we want an `in` for geographies?
https://docs.vespa.ai/en/reference/query-language-reference.html#:~:text=c%20contains%20%22C%22)-,in,-The%20in%20operator

File: src/cpr_sdk/yql_builder.py
Comment (olaughter) [inline src/cpr_sdk/yql_builder.py]: What is the thinking for using an AND here? We currently do an OR on geographies

**Review by olaughter — COMMENTED** [review]
Nice! Couple of points below, I'd also be keen to see a bit more test coverage? It'd be nice to know that given an array of geographies we get documents with those geographies, but not others

File: src/cpr_sdk/yql_builder.py
Comment (THOR300) [reply src/cpr_sdk/yql_builder.py]: Ahh right! So to confirm a filter for `["NOR", "KOR"]`  should return docs with: 
`["NOR", "KOR"]`, `["KOR"]`, `["NOR"]`

File: src/cpr_sdk/yql_builder.py
Comment (THOR300) [reply src/cpr_sdk/yql_builder.py]: Isn't document languages of the same schema (`["EN", "FR"]`) and utilises the same contain?

Comment (olaughter) [reply src/cpr_sdk/yql_builder.py]: Why, yes that should not work! 😱

File: src/cpr_sdk/yql_builder.py
Comment (olaughter) [reply src/cpr_sdk/yql_builder.py]: Correct!

File: src/cpr_sdk/yql_builder.py
Comment (olaughter) [reply src/cpr_sdk/yql_builder.py]: Well, after investigation I guess it looks like it does work ¯\_(ツ)_/¯

File: src/cpr_sdk/yql_builder.py
Comment (THOR300) [reply src/cpr_sdk/yql_builder.py]: Updated

File: tests/test_yql_builder.py
Comment (olaughter) [inline tests/test_yql_builder.py]: suggestion: Add multiple values here to make sure we are unpacking the geographies

File: tests/test_search_adaptors.py
Comment (olaughter) [inline tests/test_search_adaptors.py:543]: Love the addition of the test!

Comment (olaughter) [reply tests/test_search_adaptors.py:543]: suggestion: we also add negative condition somewhere in here making sure we aren't getting families without those geographies in them?

Comment (olaughter) [reply tests/test_search_adaptors.py:543]: nitpick: I'm finding the nested for loop difficult to parse. Especially as there are two objects being iterated (response & filters). I think there are ways to make this more readable that would be worth a moment exploring for when we revisit. One/both/other of:
- creating hits in advance with a list comprehension
- Use a dedicated function to get the attribute from the hit instead of relying on its scope within the loop

**Review by olaughter — APPROVED** [review]
Looking good! Some small test expansion I'd be keen to see before this gets merged. I've approved because I'm happy for you to make the final call on where those pieces need to be and this unblocks you to merge when you are ready. Nice work 👍

File: tests/test_search_adaptors.py
Comment (THOR300) [reply tests/test_search_adaptors.py:543]: > ...also add negative condition somewhere in here making sure we aren't getting families without those geographies in them?

`assert all([val in attribute_value_from_hit for val in filter_values])`

So this line here checks that every filter value exists in the `hit.family_geographies`. 

E.g. We filtered for `["BIH", "NOR"]`, do `"BIH"` and `"NOR"` exist in `hit.family_geographies`

So from my understanding we would throw an `AssertionError` if a family didn't have those geographies.

Comment (olaughter) [reply tests/test_search_adaptors.py:543]: Ah nice, yep that checks out 👍 thanks!

**Pull request merged**
